### PR TITLE
BUG: error in read_excel with some ods files #45598

### LINKED ISSUE #45598: BUG: Error in read_excel() function with some ods files
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
import pandas as pd

pd.read_excel('exported_with_qgis.ods')
```

### Issue Description

When taking an ods file generated from qgis (exporting a vector layer in ods format), its content.xml (internal file from ods) has newlines between XML elements, like this:
```
<table:table-row>
<table:table-cell office:value-type="string">
<text:p>CODIGO</text:p>
</table:table-cell>
...

```
instead of being like this:
```
<table:table-row><table:table-cell office:value-type="string"><text:p>CODIGO</text:p></table:table-cell>
...
```
then if I open with LibreOffice calc that file it's ok, but when doing:
```
pd.read_excel(ods_file)
```
then there is this traceback:
```
AttributeError: 'Text' object has no attribute 'qname'
```
Solution:
rewrite line no. 103 of https://github.com/pandas-dev/pandas/blob/main/pandas/io/excel/_odfreader.py
```
-sheet_cells = [x for x in sheet_row.childNodes if x.qname in cell_names]
+sheet_cells = [x for x in sheet_row.childNodes if 'qname' in dir(x) and x.qname in cell_names]
```

### Expected Behavior

The expected behaviour is to ignore the newlines. LibreOffice calc opens correctly that kind of ods

### Installed Versions

<details>

```
INSTALLED VERSIONS
------------------
commit           : 73c68257545b5f8530b7044f56647bd2db92e2ba
python           : 3.9.7.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.13.0-22-generic
Version          : #22-Ubuntu SMP Fri Nov 5 13:21:36 UTC 2021
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : es_ES.UTF-8
LOCALE           : es_ES.UTF-8

pandas           : 1.3.3
numpy            : 1.19.5
pytz             : 2021.1
dateutil         : 2.8.1
pip              : 20.3.4
setuptools       : 52.0.0
Cython           : None
pytest           : 6.0.2
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.6.3
html5lib         : 1.1
pymysql          : None
psycopg2         : 2.8.6 (dt dec pq3 ext lo64)
jinja2           : 3.0.0
IPython          : None
pandas_datareader: None
bs4              : 4.9.3
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : 3.3.4
numexpr          : None
odfpy            : None
openpyxl         : 3.0.3
pandas_gbq       : None
pyarrow          : None
pyxlsb           : None
s3fs             : None
scipy            : 1.7.3
sqlalchemy       : 1.3.22
tables           : None
tabulate         : 0.8.9
xarray           : None
xlrd             : 1.2.0
xlwt             : 1.2.0
numba            : 0.54.1

```
</details>

### PR DESCRIPTION
**Pull request by dimitra-karadima**
- [x] closes #45598 
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added an entry in the latest `doc/source/whatsnew/v1.5.0.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/io/excel/_odfreader.py [hunk 1]
```diff
@@ -102,7 +102,11 @@ def get_sheet_data(self, sheet, convert_float: bool) -> list[list[Scalar]]:
         table: list[list[Scalar]] = []
 
         for sheet_row in sheet_rows:
-            sheet_cells = [x for x in sheet_row.childNodes if x.qname in cell_names]
+            sheet_cells = [
+                x
+                for x in sheet_row.childNodes
+                if "qname" in dir(x) and x.qname in cell_names
```
Comment (phofl) [inline pandas/io/excel/_odfreader.py:108]: Can we do getattr(x, "qname")?

Comment (dimitra-karadima) [reply pandas/io/excel/_odfreader.py:108]: @phofl The tests are passing only with `if "qname" in dir(x) and getattr(x, "qname") in cell_names`. But I don't know if that's what you meant.

Comment (jreback) [reply pandas/io/excel/_odfreader.py:108]: use `hasattr` not `dir`

**Review by jreback — CHANGES_REQUESTED** [review]

**Review by jreback — APPROVED** [review]

File: pandas/tests/io/excel/test_odf.py [hunk 2]
```diff
@@ -36,3 +36,16 @@ def test_read_writer_table():
     result = pd.read_excel("writertable.odt", sheet_name="Table1", index_col=0)
 
     tm.assert_frame_equal(result, expected)
+
+
+def test_read_newlines_between_xml_elements_table():
+    # Also test reading table from an text OpenDocument file
```
Comment (phofl) [inline pandas/tests/io/excel/test_odf.py:42]: Please add gh reference with GH#number

**Review by phofl — COMMENTED** [review]
Small comment, otherwise lgtm

**Review by rhshadrach — APPROVED** [review]
lgtm - pending @phofl's request.

File: pandas/tests/io/excel/test_odf.py (hunk 2 above)
Comment (phofl) [reply pandas/tests/io/excel/test_odf.py:43]: You don‘t need the comment, the reference is enough

Comment (dimitra-karadima) [reply pandas/tests/io/excel/test_odf.py]: Done! But thanks! Good to know for my next PRs!

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
Thx @dimitra-karadima

**Pull request merged**